File: Leet Code/Merge Nodes in Between Zeros/answer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        ListNode* temp = head;
        ListNode* ans = new ListNode(0);
        ListNode* ptr = ans;

        temp = temp->next;

        int sum = 0;
        while(temp != NULL){
            while(temp->val != 0){
                sum += temp->val;
                temp = temp->next;
            }
            ptr->next = new ListNode(sum);
            ptr = ptr->next;
            temp = temp->next;
            sum = 0;
        }

        return ans->next;
    }
};

File: Leet Code/Merge Two Sorted Lists/answer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        // Checking base case
        if(list1 == null){
            return list2;
        }
        if(list2 == null){
            return list1;
        }


        ListNode temp = new ListNode();
        ListNode ans = temp;

        while( list1 != null && list2 != null){
            if(list1.val < list2.val){
                ans.next = list1;
                list1 = list1.next;
            }
            else{
                ans.next = list2;
                list2 = list2.next;
            }
            ans = ans.next;
        }

        if(list1 != null) 
            ans.next = list1;
        if(list2 != null) 
            ans.next = list2;

        return temp.next;
    }
}
